pkg/datastore/filedb/filedb.go

```diff
@@ -58,10 +76,38 @@ func (f *FileDB) Find(ctx context.Context, col datastore.Collection, opts datast
 }
 
 func (f *FileDB) Get(ctx context.Context, col datastore.Collection, id string, v interface{}) error {
-	_, ok := col.(datastore.ShardStorable)
+	fcol, ok := col.(datastore.ShardStorable)
 	if !ok {
 		return datastore.ErrUnsupported
 	}
+
+	kind := col.Kind()
+	shards := fcol.ListInUsedShards()
+	paths := make([]string, 0, len(shards))
+	for _, s := range shards {
+		paths = append(paths, makeHotStorageFilePath(kind, id, s))
+	}
+
+	parts := make([]interface{}, 0, len(paths))
+	for _, path := range paths {
+		part, err := f.fetch(ctx, col, path)
+		if err != nil {
+			f.logger.Error("failed to fetch entity",
+				zap.String("id", id),
+				zap.String("kind", kind),
+				zap.Error(err),
+			)
+			return err
+		}
+		parts = append(parts, part)
+	}
+
+	if len(parts) == 1 {
+		return dataTo(parts[0], v)
```

Comment:
I prefer to avoid using `reflect` package or copying data from another struct.
So how about unmarshal the data directly to `v`.
In the case of multiple shards, we can unmarshal the least significant one to `v` to merge.

Reply:
I feel you, if possible I would like to not use `reflect` as you said since it's slow and quite dangerous to do. 👍 

> In the case of multiple shards, we can unmarshal the least significant one to v to merge.

I get your point, but one thing I have to note have. In case of multiple shards stored object, we may need to unmarshal once to get at least `UpdatedAt` field to enable comparisons between parts and choice which should be unmarshaled first. In such a case, we may have a flow of: unmarshal => compare => unmarshal again with unmarshaled data. It looks a bit odd but could avoid using reflect, if it is good for you too, I'm happy to adopt your suggestion 👍

Reply:
I think that `UpdatedAt` does not matter while merging them. `UpdatedAt` is for the whole entity, not each field so we can not rely on it to decide the value of which entity should be used.

I think each Collection we will need a `Merge(from, to interface{}, fromShard, toShard Shard)` function that merges the `from` one to the `to` one, the shard name should be passed to that Merge function as well to know the priority of their fields.

Btw, since this PR is only supporting 1 shard currently, so let's use JSON to unmarshal directly `v` at this time for the sake of simplicity.
And then talk more about the merging problem at another PR.

Reply:
> Btw, since this PR is only supporting 1 shard currently, so let's use JSON to unmarshal directly v at this time for the sake of simplicity.
And then talk more about the merging problem at another PR.

okay, for sure 👍

Reply:
Thanks for the comment, addressed by https://github.com/pipe-cd/pipecd/pull/3286/commits/a44fd83866a60585300e80f3ea8b543b19d7eb6e 🙏